src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioVenusSpanMappingService.cs

```diff
@@ -59,6 +65,32 @@ public VisualStudioVenusSpanMappingService(VisualStudioWorkspaceImpl workspace)
             }
         }
 
+        private TextLineCollection GetTextLines(DocumentId currentDocumentId, Location location)
+        {
+            if (location.SourceTree != null)
+            {
+                // normal case - all C# and VB should hit this
+                return location.SourceTree.GetText().Lines;
+            }
+
+            var filePath = location.GetLineSpan().Path;
+            if (filePath != null)
+            {
+                // special case for typescript and etc that don't use our compilations.
+                var documentIds = _workspace.CurrentSolution.GetDocumentIdsWithFilePath(filePath);
+
+                // make sure we are dealing with right document
+                if (documentIds.Contains(currentDocumentId))
```

Comment:
I was a bit paranoid and trying to make sure given location is for the given document. for C# and VB, we filter location which doesn't belong to the document way earlier using SyntaxTree. we can do that since we can get document from syntaxtree from location.

for typescript, I wasn't sure so I was being defensive here.